### PR TITLE
Directory listing changes in Pelican client

### PR DESCRIPTION
**Pull request by joereuss12**
This is a different approach to the change of directory listings. Now, for recursive downloads, we will query the director for a PROPFIND to get the location of our "dirlisthost" (or origin). Then procede with the normal recursive download code. If we receive a 405 from the director doing this PROPFIND, we will revert back to the old way of doing things

### CODE REVIEW

File: client/handle_http.go
Comment (jhiemstrawisc) [inline client/handle_http.go:2131]: Can you construct the `job.job.directorUrl` with `url.JoinPath` to make sure it's valid? Or are we otherwise sure this construction won't result in a double `/`?

File: client/handle_http.go
Comment (jhiemstrawisc) [inline client/handle_http.go:2137]: For someone who doesn't already know what this code does, this comment won't be very elucidating. How about something like "If the director responds with 405 (method not allowed), we're working with an old Director. In that event, we try to fallback and use the deprecated dirlisthost"

File: client/handle_http.go
Comment (jhiemstrawisc) [inline client/handle_http.go:2139]: Since this is at the client, our error messages will be much more user-facing than they would be in a Pelican server like the Director. You should make the final decision, but I think we could come up with an error message that's a little more user friendly -- I don't think the average person knows what PROPFIND is or what a collections URL means.

File: client/handle_http.go
Comment (jhiemstrawisc) [inline client/handle_http.go:2151]: Does this result in duplicate error messages if we both log and return the same message?

File: client/handle_http.go
Comment (jhiemstrawisc) [inline client/handle_http.go:2188]: Same with this one, let's find a way to make this both something a user can interpret, and a developer can debug with.

File: client/handle_http.go
Comment (jhiemstrawisc) [inline client/handle_http.go:2154]: Should we also take into consideration any of the metalink header urls? For example, if the first origin from the `Location` header doesn't respond, is it worth checking any other origins that the Director indicates might serve the object?

**Review by jhiemstrawisc — CHANGES_REQUESTED** [review]

**Comment by joereuss12** [comment]
Note: this still needs a unit test to test multiple origin support so hold off on a review for now.

**Comment by joereuss12** [comment]
@bbockelm see if you like these changes better, I think I hit all of the suggestions you gave. Also, we want listings to be true by default correct?

File: server_structs/director.go
Comment (bbockelm) [inline server_structs/director.go:79]: Match the style of the other flags
```suggestion
		Listings     bool       `json:"enable_listing"`              // True if the origin allows directory listings
```

File: director/director_ui.go
Comment (bbockelm) [inline director/director_ui.go:51]: Match the fact the other flags use the singular.

```suggestion
		Listings     bool                      `json:"enableListing"`
```

File: client/fed_linux_test.go
Comment (bbockelm) [inline client/fed_linux_test.go:415]: Nitpick -- this is called uploadURL but is for downloading.  Please rename.

File: client/handle_http.go
Comment (bbockelm) [inline client/handle_http.go:1091]: Nitpick: make this a non-exported symbol at the module level (put it in the `var` block at the top of the file).  This way, we can do an `errors.Is` against it in the future.

Also, please follow go style and make error message not capitalized.

File: client/handle_http.go
Comment (bbockelm) [inline client/handle_http.go:1307]: What's this diff from?

**Review by bbockelm — APPROVED** [review]
Some minor nitpicks.  Please resolve and, once done, I'm pre-approving this to go in.

**Review by jhiemstrawisc — APPROVED** [review]
👍

**Pull request merged**
